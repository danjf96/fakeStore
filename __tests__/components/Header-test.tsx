/**
 * @format
 */

 import 'react-native';
 import React from 'react';
 import Header from '../../src/components/header';
 
 // Note: test renderer must be required after react-native.
import { fireEvent, render } from 'react-native-testing-library';
const icon = require('../../src/assets/images/logo_icon.png');

describe('renders correctly', () => {
    const onPressLeft = jest.fn()
    const onPressRight = jest.fn()
    
    const component  = <Header 
        buttonLeft={icon}
        title='TITLE'
        buttonRight={icon}

        onPressLeft={onPressLeft}
        onPressRight={onPressRight}
    />

    it('basic', () => {
        const screen = render(component);
        expect(screen.toJSON()).toMatchSnapshot()
    });

    it('more renders', () => {
        const { getAllByText, getByTestId } = render(<Header 
            buttonLeft={'LEFT'}
            title='TITLE'
            buttonRight={icon}
        />);

        expect(getAllByText(/TITLE|LEFT/)).toBeTruthy()
        expect(getByTestId('imageRight')).toBeTruthy()
       
    });

    it('functions', () => {

        const { getByTestId } = render(component);

        expect(getByTestId('imageLeft')).toBeTruthy()

        fireEvent(getByTestId('buttonLeft'), 'onPress')
        expect(onPressLeft).toBeCalled()

        fireEvent(getByTestId('buttonRight'), 'onPress')
        expect(onPressRight).toBeCalled()
    });
})

 