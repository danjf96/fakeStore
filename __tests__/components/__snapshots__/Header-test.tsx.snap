// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly basic 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "width": "100%",
    }
  }
>
  <View
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="buttonLeft"
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/images/logo_icon.png",
        }
      }
      style={
        Object {
          "height": 18.5,
          "resizeMode": "contain",
          "width": 18.33,
        }
      }
      testID="imageLeft"
    />
  </View>
  <Text
    style={
      Object {
        "color": "#85868A",
        "fontFamily": "WorkSans-Regular",
        "fontSize": 14,
        "fontWeight": "700",
        "textTransform": "uppercase",
      }
    }
  >
    TITLE
  </Text>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="buttonRight"
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/images/logo_icon.png",
        }
      }
      style={
        Object {
          "height": 30,
          "resizeMode": "contain",
          "width": 30,
        }
      }
      testID="imageRight"
    />
  </View>
</View>
`;
